src/web/src/pages/CourseExplorer.vue

```diff
@@ -55,12 +103,37 @@ export default {
     return {
       //an object with keys being the dept, and values a list of courses
       deptClassDict: {},
+      schoolsMajorDict: {},
       ready: false,
+
+      /*
+       *Used to define the order that the schools are placed into the two main
+       *columns. This could be implemented generically, but making the columns
+       *more or less "even" is NP-Hard so its more convenient to just define
+       *manually
+       */
+      schoolOrder: [
+        "Engineering",
+        "Science",
+        "Humanities, Arts and Social Sciences",
+        "Architecture",
+        "Business Management",
+        "Other",
+      ],
+      courseInfoModalCourse: null,
+      showCourseInfoModal: false,
     };
   },
   async created() {
     getCourses(this.selectedSemester).then((courses) => {
       for (const c of courses) {
+        if (!this.schoolsMajorDict[c.school]) {
+          this.schoolsMajorDict[c.school] = [c.department];
+        } else {
+          if (!this.schoolsMajorDict[c.school].includes(c.department)) {
```

Comment:
could use javascript Set instead of list, then you can just do `this.schoolsMajorDict[c.school].add(c.department)`